torch/_inductor/runtime/benchmarking.py

```diff
@@ -0,0 +1,1026 @@
+        # with an absolute minimum of 1
+        reduced_benchmark_iters = max(min(benchmark_iters, reduced_benchmark_iters), 1)
+        return reduced_benchmark_iters
+
+    def get_required_gpu_sleep_cycles(
+        self: Self,
+        memory_warmup_iters: int,
+        benchmark_iters: int,
+        cpu_launch_overhead_ms_per_iter: float,
+    ) -> int:
+        cpu_launch_overhead_ms_for_memory_warmup = (
+            memory_warmup_iters * self.cpu_launch_overhead_ms_per_gpu_cache_clear
+        )
+        cpu_launch_overhead_ms_for_benchmarking = (
+            benchmark_iters * cpu_launch_overhead_ms_per_iter
+        )
+        total_cpu_launch_overhead_ms = (
+            cpu_launch_overhead_ms_for_memory_warmup
+            + cpu_launch_overhead_ms_for_benchmarking
+        )
+        # we want the GPU sleep to overlap the CPU launch overhead of
+        # the memory warmup only, and not the GPU time. we can and should
+        # have the actual memory warmup begin on the GPU before we finish
+        # queueing the benchmark iterations because we are not timing the
+        # memory warmups and can therefore save on some total duration
+        required_gpu_sleep_duration_ms = total_cpu_launch_overhead_ms - (
+            self.gpu_time_ms_per_gpu_cache_clear * memory_warmup_iters
+        )
+        required_gpu_sleep_cycles = max(
+            int(required_gpu_sleep_duration_ms / self.gpu_time_ms_per_gpu_clock_cycle),
+            0,
+        )
+        return required_gpu_sleep_cycles
+
+    def get_event_pairs(
+        self: Self, num_pairs: int
+    ) -> List[Tuple[torch.cuda.Event, torch.cuda.Event]]:
+        return [
+            (
+                torch.cuda.Event(enable_timing=True),
+                torch.cuda.Event(enable_timing=True),
+            )
+            for _ in range(num_pairs)
+        ]
+
+    def get_interleaved_event_pairs(
+        self: Self, num_callables: int, num_pairs: int
+    ) -> List[List[Tuple[torch.cuda.Event, torch.cuda.Event]]]:
+        return [self.get_event_pairs(num_callables) for _ in range(num_pairs)]
+
+    def get_min_timing_ms(
+        self: Self, event_pairs: List[Tuple[torch.cuda.Event, torch.cuda.Event]]
+    ) -> float:
+        return min(
+            [
+                start_event.elapsed_time(end_event)
+                for start_event, end_event in event_pairs
+            ]
+        )
+
+    def get_interleaved_min_timings_ms(
+        self: Self,
+        interleaved_event_pairs: List[List[Tuple[torch.cuda.Event, torch.cuda.Event]]],
+    ) -> List[float]:
+        return [
+            self.get_min_timing_ms(list(event_pairs))
+            for event_pairs in zip(*interleaved_event_pairs)
+        ]
+
+
+benchmarker = Benchmarker()
+
+
+def do_bench(callable_or_callables: Union[Callable[..., Any], List[Callable[..., Any]]], *args: Any, lazy: bool = False, **kwargs: Any) -> Union[LazyBenchmark, float]:
```

Comment:
@bertmaher Added this do_bench wrapper, its not perfect (mainly because we have no way of differentiating CPU and GPU callables once they've been wrapped to be callable by the benchmarker) but it does reduce nominal changes.

Reply:
Eh actually I don't like how this silently dispatches to GPU since it can't differentiate the devices, I'm going to revert the change. It does result in more nominal changes like you mentioned, but I think it is worth having the clarity of knowing what is called.